### ISSUE TITLE
TextInput linked to waveform breaks on input

### ISSUE THREAD

**Issue by Tini4** [React-Automation-Studio/React-Automation-Studio#119]
I have a TextInput component that receives a waveform PV with character array:
```js
<TextInput
  pv="$(DEVICE):RawTiff:FilePath"
  macros={{ "$(DEVICE)": props.device }}
  label={"Save path"}
  useStringValue={true}
  debug={true}
/>
```
![image](https://github.com/React-Automation-Studio/React-Automation-Studio/assets/67708027/97be8827-9ea3-4279-b387-105881ebb862)
When I try to to change the value (lets say to /tempa), this happens:
![image](https://github.com/React-Automation-Studio/React-Automation-Studio/assets/67708027/70bed753-f503-4de8-8bb3-e48242b105f1)
And I get this error in the console:
```
Warning: A component is changing a controlled input to be uncontrolled. This is likely caused by the value changing from a defined to undefined, which should not happen. Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://reactjs.org/link/controlled-components
```
The UI breaks, and the value gets set to `undefined`.
When looking at the EPICS PV with camonitor, the PV gets set correctly, so the problem seems to just be in the React/UI part.
Displaying the PV works fine, but something goes wrong when updating it.
The TextOutput component holding the same PV also breaks the same way, when you try to update the value with TextInput:
![image](https://github.com/React-Automation-Studio/React-Automation-Studio/assets/67708027/4968bf7c-6354-4f9a-921e-d15c1f330dd7)
On the left is TextInput where I try to change the value and on the right there is TextOutput that also breaks.

**Comment by wduckitt** [comment]
Hi, 

As mentioned before, you are trying to edit a waveform Epics PV as a pure string, with a TextInput. This has never been the intended use case. The TextInput has no knowedge of the array length, format or indices of the array waveform. Somewhere something breaks when you do this. For me it appears to be an anitpattern use case of the waveform PV in the AreaDetector module that you are trying to create a UI for. This implies that somehow for other user interface systems such as medm there is a hack for this specific usecase as I see the areadetector module is documented primarily using medm that we re not aware of.

There are two possible options:

1. write the string value to stringout PV, and tie the OUT to the waveform of bytes...
2. We debug the situation, but for this I need better insight into the use case. Can you please fork the project, and create a simple UI to illustrate the problem, you will need to add in an EPICS test variable wavform into RAS's testIOC that producesthe same problem. https://github.com/React-Automation-Studio/React-Automation-Studio/blob/master/epics/testIOC/db/test.db

If you share the forked repo with the test scenario with me, I can debug, and maybe create a workaround.

**Comment by wduckitt** [comment]
similar to #117

**Comment by Tini4** [comment]
There you go: [fork](https://github.com/Tini4/React-Automation-Studio.git).
I created a UI with the error on route /Example4.

**Comment by wduckitt** [comment]
I have added a patch in pull request #118, it was a simple fix, made easy by debugging the fork with the example you provided.

**Comment by wduckitt** [comment]
issue was closed by #118 and is active in the master branch to test.
